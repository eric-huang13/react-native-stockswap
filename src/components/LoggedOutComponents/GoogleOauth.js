import React, {useEffect} from 'react';
import {StyleSheet, View, Text, TouchableOpacity, Image} from 'react-native';
import {connect} from 'react-redux';
import GoogleIcon from '../../icons/GoogleIcon';
import {moderateScale} from '../../util/responsiveFont';
import {GoogleLogin, GoogleLogout, GoogleIsSignedIn, GoogleSignUp} from '../../actions/user';

export function GoogleOauth(props) {
  console.log(props,'props google')
  useEffect(() => {
    props.GoogleIsSignedIn();
  }, []);

  const signIn = () => {
    props.signup ? 
    props.GoogleSignUp() :
    props.GoogleLogin()
   
  };

  const signOut = () => {
    props.GoogleLogout();
  };
  return (
    <View>
      {props.googleUser.idToken && props.userData.accessToken ? (
       <TouchableOpacity onPress={signOut}>
       <View style={styles.alternateSignupInner}>
         <View style={styles.signupIcon}>
           <Image
             style={{width: 23, height: 23}}
             source={{uri: props.googleUser.user.photo}}
           />
         </View>          
           <Text style={styles.alternateSignUpButton}>LOGOUT OF GOOGLE</Text>          
       </View>
       </TouchableOpacity>
      ) : (
        <TouchableOpacity onPress={signIn}>
        <View style={styles.alternateSignupInner}>
          <View style={styles.signupIcon}>
            <GoogleIcon />
          </View>
          {props.signup ? 
          <Text style={styles.alternateSignUpButton}>
              SIGN UP WITH GOOGLE
            </Text>  :
            <Text style={styles.alternateSignUpButton}>
              LOGIN WITH GOOGLE
            </Text> 
}          
                     
        </View>
        </TouchableOpacity>
     
      )}
    </View>
  );
}
const mapStateToProps = (state) => {
  return {
    googleUser: state.user.googleUser,
    userData: state.user.userData,

  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    GoogleSignUp: () => dispatch(GoogleSignUp()),
    GoogleLogin: () => dispatch(GoogleLogin()),
    GoogleLogout: () => dispatch(GoogleLogout()),
    GoogleIsSignedIn: () => dispatch(GoogleIsSignedIn()),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(GoogleOauth);

const styles = StyleSheet.create({
  alternateSignUpButton: {
    alignSelf: 'center',
    justifyContent: 'center',
    color: '#FFFFFF',
    textAlign: 'center',
    fontSize: moderateScale(14),
    fontFamily: 'Montserrat-SemiBold',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 1,
  },
  alternateSignupInner: {
    alignSelf: 'center',
    paddingVertical: moderateScale(12),
    paddingHorizontal: moderateScale(4),
    backgroundColor: '#2C3957',
    width: moderateScale(350),
    borderRadius: moderateScale(8),
    marginBottom: moderateScale(10),
    flexDirection: 'row',
  },
  signupIcon: {
    padding: moderateScale(7),
    backgroundColor: '#3A4A6D',
    borderRadius: moderateScale(7),
    marginVertical: moderateScale(-8),
    marginRight: moderateScale(63),
    alignSelf: 'center',
  },
});
