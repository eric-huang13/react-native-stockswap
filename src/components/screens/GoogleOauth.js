import React, {useEffect, useState} from 'react';
import {StyleSheet, View, Text, TouchableOpacity, Image} from 'react-native';
import {
  GoogleSignin,
  GoogleSigninButton,
  statusCodes,
} from '@react-native-community/google-signin';
import GoogleIcon from '../../icons/GoogleIcon';
import GoogleLogin from './GoogleLogin'

export default function GoogleOauth() {
    const [user, setUser] = useState({});
  useEffect(() => {
    GoogleSignin.configure({
      webClientId:
        '534509051413-6a8ceait2pji394mgui3svtrnp7bl4hp.apps.googleusercontent.com',
      offlineAccess: true, // if you want to access Google API on behalf of the user FROM YOUR SERVER

      //need to figure out way for this screen to show
      // forceCodeForRefreshToken: true, // [Android] related to `serverAuthCode`.

      iosClientId: '', // [iOS] optional, if you want to specify the client ID of type iOS (otherwise, it is taken from GoogleService-Info.plist)
    });
    isSignedIn();
  }, []);
  const signIn = async () => {
    try {
      await GoogleSignin.hasPlayServices();
      const userInfo = await GoogleSignin.signIn();
      console.log(userInfo, 'info signin');
      setUser(userInfo);
      //send to backend
      // RegisterUser({token:userInfo.idToken, email:userInfo.user.email});
    } catch (error) {
      console.log('Message', error.message);
      if (error.code === statusCodes.SIGN_IN_CANCELLED) {
        console.log('User Cancelled the Login Flow');
      } else if (error.code === statusCodes.IN_PROGRESS) {
        console.log('Signing In');
      } else if (error.code === statusCodes.PLAY_SERVICES_NOT_AVAILABLE) {
        console.log('Play Services Not Available or Outdated');
      } else {
        console.log('Some Other Error Happened');
      }
    }
  };
  const isSignedIn = async () => {
    const isSignedIn = await GoogleSignin.isSignedIn();
    if (isSignedIn) {
      getCurrentUserInfo();
    } else {
      console.log('Please Login');
    }
  };

  const getCurrentUserInfo = async () => {
    try {
      const userInfo = await GoogleSignin.signInSilently();
      setUser(userInfo);
    } catch (error) {
      if (error.code === statusCodes.SIGN_IN_REQUIRED) {
        console.log('User has not signed in yet');
      } else {
        console.log("Something went wrong. Unable to get user's info");
      }
    }
  };
  const signOut = async () => {
    try {
      await GoogleSignin.revokeAccess();
      await GoogleSignin.signOut();
      setUser({}); // Remember to remove the user from your app's state as well
    } catch (error) {
      console.error(error);
    }
  };
  // s
    return (
            <GoogleLogin signIn={signIn} signOut={signOut} user={user}/>
           
    )
}
 