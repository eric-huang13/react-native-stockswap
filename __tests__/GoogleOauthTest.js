import React from 'react';
import renderer from 'react-test-renderer';
import {GoogleOauth} from '../src/components/LoggedOutComponents/GoogleOauth'
import ShallowRenderer from 'react-test-renderer/shallow'; // ES6


jest.mock('@react-native-community/google-signin', () => {});

jest.mock("@react-native-async-storage/async-storage", () =>
  require("@react-native-async-storage/async-storage/jest/async-storage-mock"),
);


test('renders correctly', () => {
  const renderer = new ShallowRenderer();
  let googleUser={idToken:'test',
                user:{photo:'test'}
                }
  renderer.render(<GoogleOauth googleUser={googleUser}/>);
  const tree = renderer.getRenderOutput();
  expect(tree).toMatchSnapshot();
});