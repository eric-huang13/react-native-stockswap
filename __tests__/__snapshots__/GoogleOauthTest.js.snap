// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <ForwardRef
    onPress={[Function]}
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#2C3957",
          "borderRadius": 11.299270072992702,
          "flexDirection": "row",
          "marginBottom": 14.124087591240876,
          "paddingHorizontal": 5.649635036496351,
          "paddingVertical": 16.94890510948905,
          "width": 494.3430656934307,
        }
      }
    >
      <View
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#3A4A6D",
            "borderRadius": 9.886861313868614,
            "marginRight": 88.98175182481751,
            "marginVertical": -11.299270072992702,
            "padding": 9.886861313868614,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "test",
            }
          }
          style={
            Object {
              "height": 23,
              "width": 23,
            }
          }
        />
      </View>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#FFFFFF",
            "elevation": 1,
            "fontFamily": "Montserrat-SemiBold",
            "fontSize": 19.77372262773723,
            "justifyContent": "center",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 3.84,
            "textAlign": "center",
          }
        }
      >
        LOGOUT OF GOOGLE
      </Text>
    </View>
  </ForwardRef>
</View>
`;
